aten/src/ATen/native/cuda/EmbeddingBag.cu

```diff
@@ -141,78 +163,25 @@ __global__ void EmbeddingBag_accGradParametersKernel(
     } while (idx < numel && input[idx] == input[idx - 1]);
   }
 }
-}
-
-std::tuple<Tensor, Tensor, Tensor>
-embedding_bag_cuda(const Tensor &weight, const Tensor &indices,
-                   const Tensor &offsets, const bool scale_grad_by_freq,
-                   const int64_t mode, bool sparse) {
-  auto indices_arg = TensorArg(indices, "indices", 1);
-  checkScalarType("embedding_bag_cuda", indices_arg, kLong);
-  checkContiguous("embedding_bag_cuda", indices_arg);
-  auto offsets_arg = TensorArg(offsets, "offsets", 1);
-  checkScalarType("embedding_bag_cuda", offsets_arg, kLong);
-  checkContiguous("embedding_bag_cuda", offsets_arg);
-  auto weight_arg = TensorArg(weight, "weight", 1);
-  checkContiguous("embedding_bag_cuda", weight_arg);
-  checkSameGPU("embedding_bag_cuda", weight_arg, indices_arg);
-  checkSameGPU("embedding_bag_cuda", weight_arg, offsets_arg);
-
-  int64_t numIndices = indices.sizes()[0];
-  int64_t numBags = offsets.sizes()[0];
-  int64_t stride = weight.sizes()[1];
-
-  auto bag_size = at::zeros(indices.type(), offsets.sizes());
-  auto offset2bag =
-      at::zeros(indices.type(), {indices.sizes()[0]}); // offset2bag = [0 0 0 0 0]
-
-  cudaStream_t stream = globalContext().getCurrentCUDAStream();
-
-  auto output = at::zeros(weight.type(), {offsets.sizes()[0], weight.sizes()[1]});
-
-  dim3 block = dim3(32, 8);
-  int grid = 1024;
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(weight.type(), "embedding_bag_cuda", [&] {
-    using cuda_scalar_t = cuda::type<scalar_t>;
-    EmbeddingBag_updateOutputKernel<cuda_scalar_t><<<grid, block, 0, stream>>>(
-        indices.data<int64_t>(), offsets.data<int64_t>(),
-        weight.data<cuda_scalar_t>(), output.data<cuda_scalar_t>(),
-        offset2bag.data<int64_t>(), numIndices, numBags, stride, mode,
-        bag_size.data<int64_t>());
-  });
 
-  THCudaCheck(cudaGetLastError());
-  return std::tuple<Tensor, Tensor, Tensor>(output, offset2bag, bag_size);
-}
 
-Tensor embedding_bag_backward_cuda(const Tensor &grad_, const Tensor &indices,
-                                   const Tensor &offsets,
+Tensor embedding_bag_backward_cuda_avg_sum(
```

Comment:
it's really confusing to have `embedding_bag_backward_cuda_avg_sum` and `embedding_bag_backward_cuda_sum_avg` in the same file, one being internal to the other. You should call it `embedding_bag_backward_cuda_sum_avg` and `embedding_bag_backward_cuda_sum_avg_kernel`

Reply:
Done.